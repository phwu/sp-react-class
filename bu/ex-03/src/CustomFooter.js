import React from 'react';

const Footer = (props) => {
    return (
        <div>
        <p>Copyright &copy; {new Date().getFullYear()} {props.company.name}</p>
            <address>
        {props.company.address} <br/>
            {props.company.city}, {props.company.state}<br/>
            {props.company.zip}
            </address>
        </div>
    );
};

export default Footer;