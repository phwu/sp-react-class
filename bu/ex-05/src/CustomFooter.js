import React from 'react';

const CustomFooter = ( props ) => {
  return (
    <div className="footer">
      <hr/>
      <p>Copyright &copy; 2017 {props.company.name}</p>
      <address>
        {props.company.address}<br/>
        {props.company.city}, {props.company.state}<br/>
        {props.company.zip}
      </address>
    </div>
  );
}

export default CustomFooter;
