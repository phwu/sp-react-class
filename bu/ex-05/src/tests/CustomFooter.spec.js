import React from 'react';
import renderer from 'react-test-renderer';

import CustomFooter from '../CustomFooter';

it('Testing snapshot', () => {
    const myCompany = {
        name: 'Polly rocks',
        address: '123',
        state: 'FL'
    };

    const snap = renderer.create(
        <CustomFooter company={myCompany}/>
        ).toJSON();

    expect(snap).toMatchSnapshot();
});