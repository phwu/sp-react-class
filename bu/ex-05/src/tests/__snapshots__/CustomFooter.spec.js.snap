// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing snapshot 1`] = `
<div
  className="footer"
>
  <hr />
  <p>
    Copyright Â© 2017 
    Polly rocks
  </p>
  <address>
    123
    <br />
    , 
    FL
    <br />
  </address>
</div>
`;
